LibTest/js_interop/JS/js_A03_t02.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion An annotation on a JavaScript interop declaration.
+/// ...
+/// Specifying name customizes the JavaScript name to use, which can be used in
+/// the following scenarios:
+/// - Adding a JavaScript prefix to all the external top-level declarations,
+///   static members, and constructors of a library by parameterizing the
+///   annotation on the library with `name`.
+/// - Specifying the JavaScript class to use for external static members and
+///   constructors of an interop extension type by parameterizing the annotation
+///   on the interop extension type with `name`.
+/// - Renaming external declarations by parameterizing the annotation on the
+///   member with `name`.
+///
+/// @description Check that specifying `name` customizes the JavaScript `name`
+/// to use. Test extension type declaration.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../js_utils.dart';
+
+@JS("JSEt")
+extension type ET(JSObject _) implements JSObject {
+  external String foo();
```

Comment:
nit: Perhaps an `external` constructor and factory so we know that's correctly lowered?

Reply:
Interesting! I'm not sure where we'd find anything in JavaScript which will faithfully emulate a factory constructor! So how is it even possible to lower that? ;-)

Reply:
@eernstg I believe it is described here https://dart.dev/interop/js-interop/usage#interop-type-members. 

I'm going to add tests for constructors in a separate PR. As far as understand these constructirs perfectly work without `@JS()` annotation, therefore there will be tests for "general interop", not for `@JS()` annotation.

Reply:
Aha, "same semantics .. regardless whether they're a factory". So they are essentially not lowered to "a factory constructor" in JavaScript, the keyword `factory` in the Dart interop code is simply ignored. Makes sense (or it could be a warning because ignoring what a developer writes may create misinterpretations of the situation).

But that will be handled later on, as you say.